typescript/cli/src/config/agent.ts

```diff
@@ -19,19 +23,80 @@ export async function createAgentConfig({
   out,
 }: {
   context: CommandContext;
-  chains: string[];
+  chains?: string[];
   out: string;
 }) {
   logBlue('\nCreating agent config...');
 
-  const { registry, multiProvider, chainMetadata } = context;
+  const { registry, multiProvider, chainMetadata, skipConfirmation } = context;
   const addresses = await registry.getAddresses();
 
-  const core = HyperlaneCore.fromAddressesMap(addresses, multiProvider);
+  await handleNoChainsProvided(skipConfirmation, chains);
 
-  const startBlocks = await promiseObjAll(
-    objMap(addresses, async (chain, _) => {
-      // If the index.from is specified in the chain metadata, use that.
+  const chainAddresses = filterChainAddresses(addresses, chains);
+
+  const core = HyperlaneCore.fromAddressesMap(chainAddresses, multiProvider);
+  const startBlocks = await getStartBlocks(chainAddresses, core, chainMetadata);
+
+  if (!skipConfirmation) {
+    await handleMissingInterchainGasPaymaster(chainAddresses);
```

Comment:
This is probably okay but my expectation would have been to pass `skip` into `handleMissingInterchainGasPaymaster` and have it fail if it's both not provided and skip is true. See https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/rossy/cli-warp-output-shape/typescript/cli/src/deploy/core.ts#L48